src/CheckOut.js

```diff
@@ -161,14 +163,32 @@ class CheckOut extends React.Component {
     this.state = { loading: false, blocked: false };
     this.patronFormRef = React.createRef();
     this.timer = undefined;
+    this.shouldSubmitOnMount = hasIn(location, 'state.patronBarcode') && hasIn(location, 'state.itemBarcode');
     this.state = { submitting: false };
   }
 
   componentDidMount() {
-    this.patronFormRef.current.focus();
+    const {
+      setFieldValue,
+      location,
+    } = this.props;
+
+    if (this.shouldSubmitOnMount) {
+      setFieldValue('patronForm', 'patron.identifier', location.state.patronBarcode);
+      setFieldValue('itemForm', 'item.barcode', location.state.itemBarcode);
+    } else {
+      this.patronFormRef.current.focus();
+    }
   }
 
   componentDidUpdate(prevProps) {
+    const { submitForm } = this.props;
+    const patronFormFilled = !prevProps.patronBarcodeFieldValue && this.props.patronBarcodeFieldValue;
+
+    if (patronFormFilled && this.shouldSubmitOnMount) {
```

Comment:
Prease rename shouldSubmitOnMount to more meaningful as no submit on mount happened